tests/test_detectAndMeasure.py

```diff
@@ -431,6 +431,87 @@ def _detection_wrapper(setFlags=True):
+        This is testing method called updateMasks
+        """
+        xSize = 256
+        ySize = 256
+        science, sources = makeTestImage(psfSize=2.4, xSize=xSize, ySize=ySize, doApplyCalibration=True)
+        science_fake_img, science_fake_sources = makeTestImage(
+            psfSize=2.4, xSize=xSize, ySize=ySize, seed=5, nSrc=3, noiseLevel=0.25, fluxRange=1
+        )
+        template, _ = makeTestImage(psfSize=2.4, xSize=xSize, ySize=ySize, doApplyCalibration=True)
+        tmplt_fake_img, tmplt_fake_sources = makeTestImage(
+            psfSize=2.4, xSize=xSize, ySize=ySize, seed=9, nSrc=3, noiseLevel=0.25, fluxRange=1
+        )
+        # created fakes and added them to the images
+        science.image.array += science_fake_img.image.array
+        template.image.array += tmplt_fake_img.image.array
+
+        # TODO: DM-40796 update to INJECTED names when source injection gets refactored
+        # adding mask planes to both science and template images
+        science.mask.addMaskPlane("FAKE")
+        science_fake_bitmask = science.mask.getPlaneBitMask("FAKE")
+        template.mask.addMaskPlane("FAKE")
+        template_fake_bitmask = template.mask.getPlaneBitMask("FAKE")
+
+        for a_science_source in science_fake_sources:
+            # 3 x 3 masking of the source locations is fine
+            bbox = lsst.geom.Box2I(
+                lsst.geom.Point2I(a_science_source.getX(), a_science_source.getY()), lsst.geom.Extent2I(3, 3)
+            )
+            science[bbox].mask.array |= science_fake_bitmask
+
+        for a_template_source in tmplt_fake_sources:
+            # 3 x 3 masking of the source locations is fine
+            bbox = lsst.geom.Box2I(
+                lsst.geom.Point2I(a_template_source.getX(), a_template_source.getY()),
+                lsst.geom.Extent2I(3, 3)
+            )
+            template[bbox].mask.array |= template_fake_bitmask
+
+        science_fake_masked = (science.mask.array & science_fake_bitmask) > 0
+        template_fake_masked = (template.mask.array & template_fake_bitmask) > 0
+
+        subtractConfig = subtractImages.AlardLuptonSubtractTask.ConfigClass()
+        subtractTask = subtractImages.AlardLuptonSubtractTask(config=subtractConfig)
+        subtraction = subtractTask.run(template, science, sources)
+
+        # check subtraction mask plane is set where we set the previous masks
+        diff_mask = subtraction.difference.mask
+
+        # science mask should be now in INJECTED
+        inj_masked = (diff_mask.array & diff_mask.getPlaneBitMask("INJECTED")) > 0
+
+        # template mask should be now in INJECTED_TEMPLATE
+        injTmplt_masked = (diff_mask.array & diff_mask.getPlaneBitMask("INJECTED_TEMPLATE")) > 0
+
+        self.assertEqual(np.sum(inj_masked.astype(int)-science_fake_masked.astype(int)), 0)
+        self.assertEqual(np.sum(injTmplt_masked.astype(int)-template_fake_masked.astype(int)), 0)
```

Comment:
For these two asserts, use `self.assertMasksEqual(inj_masked.astype(int), science_fake_masked.astype(int))`, which will give a more helpful error message if they are different.
I don't think there is an assert that checks arrays of booleans, so I think you do have to still do the `.astype(int)`. The tests pass without it, but if the arrays are actually different then the assert itself causes an error, which is unhelpful!
Note that in this case you could use `self.assertMasksEqual ` as I suggest, or `self.assertImagesEqual` or `assertFloatsEqual`

Reply:
Chose `assertFloatsEqual` in this case. 
I have followed your suggestion. Thanks